### PR TITLE
Gentoo & Pidfile/Init fixes

### PR DESCRIPTION
**Pull request by robbat2**
This started as writing a Gentoo init script, but ended up being fixing pidfiles so the init scripts could work properly.

### CODE REVIEW

**Comment by robbat2** [comment]
Hmm, this might have introduced a subtle bug somewhere, and there isn't a testsuite for the webhook:
[2015-05-14 08:16:24] ERROR TypeError: no implicit conversion of Array into Integer
	/usr/lib64/ruby/2.0.0/openssl/ssl.rb:278:in `initialize'

**Comment by robbat2** [comment]
Those trailing commas from me merging were the root of that webhook breakage.

File: templates/webhook.init.erb [hunk 1]
```diff
@@ -41,15 +41,9 @@ start() {
...
-    pid=`netstat -lnp | awk  '/.*:8088.*ruby/{sub("/.*","",$NF);print $NF}'`
-    #killproc -p $pidfile $webhook]
-    if kill -9 $pid ; then
-      success 'webhook'
-      return 0
-    else
-      failure 'webhook'
-      return 1
-    fi
+    killproc -p $pidfile $webhook
```
Comment (acidprime) [inline templates/webhook.init.erb]: So this causes upgrade issues as the running demon does not have a pid

**Comment by acidprime** [comment]
Also seeing 
```
service webhook status
/sbin/service: line 66: 12178 Segmentation fault      (core dumped) env -i PATH="$PATH" TERM="$TERM" "${SERVICEDIR}/${SERVICE}" ${OPTIONS}
```

File: templates/webhook.init.erb (hunk 1 above)
Comment (robbat2) [inline templates/webhook.init.erb]: Hmm, how can we forcibly restart it just this once, so it has a pid in future?

**Comment by robbat2** [comment]
I don't get the segfault on either Ubuntu 14.04 or Gentoo.

**Comment by acidprime** [comment]
This is under Centos, trying to track it down

**Pull request merged**